src/main/java/com/google/api/codegen/transformer/GapicMethodContext.java

```diff
@@ -97,4 +91,19 @@ public GapicMethodContext cloneWithEmptyTypeTable() {
         getFlatteningConfig(),
         getFeatureConfig());
   }
+
+  @Override
+  public String getAndSaveRequestTypeName() {
+    return getTypeTable().getAndSaveNicknameFor(getMethod().getInputType());
+  }
+
+  @Override
+  public String getAndSaveResponseTypeName() {
+    return getTypeTable().getAndSaveNicknameFor(getMethod().getOutputType());
+  }
+
+  @Override
+  public String getBatchingDescriptorConstName() {
+    return getNamer().getBatchingDescriptorConstName(getMethod());
+  }
```

Comment:
Yeah, moving to the yet-to-be-created abstracted Method interface would be better.